CRM/MembershipExtras/Hook/Pre/ContributionRecur.php

```diff
@@ -66,18 +67,18 @@ private function getRecurringContribution($id) {
    * contribution.
    */
   public function preProcess() {
-    if ($this->operation == 'create') {
+    $isPaymentPlanPayment = CRM_MembershipExtras_Helper_InstalmentSchedule::isPaymentPlanPayment();
```

Comment:
You shouldn't do this here, `CRM_MembershipExtras_Helper_InstalmentSchedule::isPaymentPlanPaymen()` checks for some form elements if they are set, where this hook is a database level hook and might run from various places at different times.

Reply:
@omarabuhussein  I understood that only need to process this if the payment plan was created to the UI and after our conversion to slack. I removed the code and created this ticket, so we do not forget that we would need to deal with it as it is still causing failure in GoCardless https://compucorp.atlassian.net/browse/CIWEMB-335

Reply:
yeah, all I am saying here is that the approach used here is not the right one to solve the problem and might cause issues in some other places if we happen to proceed with it, so we need to come up with something else.